File: kmm/composeApp/src/commonMain/kotlin/com/danioliveira/taskmanager/domain/usecase/tasks/UpdateTaskStatusUseCase.kt
package com.danioliveira.taskmanager.domain.usecase.tasks

import com.danioliveira.taskmanager.api.response.TaskResponse
import com.danioliveira.taskmanager.domain.TaskStatus
import com.danioliveira.taskmanager.domain.repository.TaskRepository

/**
 * Use case for updating task status.
 *
 * @property taskRepository The repository for task operations
 */
class UpdateTaskStatusUseCase(private val taskRepository: TaskRepository) {

    /**
     * Updates the status of a task.
     *
     * @param taskId The ID of the task to update
     * @param status The new status of the task
     * @return Result containing the updated task details
     */
    suspend operator fun invoke(taskId: String, status: TaskStatus): Result<TaskResponse> {
        return taskRepository.updateTaskStatus(taskId, status.name)
    }
} 

File: task_manager_dart/app/lib/features/tasks/domain/usecases/update_task_status_usecase.dart
import 'package:task_manager_shared/models.dart';
import '../../../../core/utils/result.dart';
import '../../data/repositories/task_repository.dart';

/// Use case for updating task status
/// Matches KMM's UpdateTaskStatusUseCase
class UpdateTaskStatusUseCase {
  final TaskRepository _repository;

  UpdateTaskStatusUseCase(this._repository);

  Future<Result<TaskDto>> call(String taskId, TaskStatus status) async {
    return await _repository.changeTaskStatus(taskId, status);
  }
}

